library/math/matrix_related/binary_matrix_mult.hpp

```diff
@@ -0,0 +1,21 @@
+/** @file */
+#pragma once
+/**
+ * @see https://codeforces.com/blog/entry/80195
+ *
+ * @param a,b binary matrices
+ * @returns a*b mod 2
+ * @time O(n * m * inner / w) where w is the machine word size (usually 32 or
+ * 64)
+ * @space besides the O(n * inner) and O(inner * m) params, this function
+ * allocates and returns a O(n * m) vector
+ */
+template <unsigned long int N>
```

Comment:
actually thinking it would be nice to support N, M, K all here for wide and long matrices